examples/nextjs-with-typescript/pages/MuxPlayer.tsx

```diff
@@ -268,6 +259,24 @@ const getControlCustomizationCSSVars = (state) => {
     .map(([k]) => k);
 };
 
+type Props = { location?: Pick<Location, 'origin' | 'pathname'> };
+
+const getUrl = ({ req, resolvedUrl }) => {
+  const { headers } = req;
+  const refererUrl = headers.referer && new URL(headers.referer);
+  const baseUrlHost = headers.host.toLowerCase();
+  const refererHost = refererUrl?.host?.toLowerCase();
+
+
+  if (refererHost === baseUrlHost && headers["sec-fetch-site"] === "same-origin") {
+    return new URL(refererUrl?.origin ? refererUrl.origin + resolvedUrl : './');
```

Comment:
the main change in this method is making sure that `resolvedUrl` is always part of the url, because otherwise, there are cases where `/MuxPlayer` doesn't make it in the url